### PR TITLE
Add Support for Sequential Execution(For Single Test bed/workload)

### PR DESCRIPTION
**Pull request by DhobaleVishal**
## Problem Statement

-   Added support for sequential execution(for single test bed/workload)

## Design

-   For Bug, Describe the fix here.
-   For new test addition post the link for design

## Coding

   Checklist for Author

-   [x]  Coding conventions are followed and code is formatted (e.g. using pycharm in-built formatter)

## Testing

  Checklist for Author

-   [x]  New/Affected tests are executed
-   [x]  Attach test execution logs
https://jts.seagate.com/secure/attachment/526693/CORTX-32756_sequential_execution_build_2.0.0-845-07604.zip

## Review 

  Checklist for Author

-   [x]  JIRA number/GitHub Issue added to PR
-   [x]  PR is self reviewed
-   [x]  Jira and state/status is updated and JIRA is updated with PR link
-   [x]  Check if the description is clear and explained

## Documentation

  Checklist for Author

-   [x]  Changes done to ReadMe

-----
[View rendered README.md](https://github.com/DhobaleVishal/corio/blob/CORTX-32756_support_for_sequential_execution_for_single_test_bed/workload/README.md)

### CODE REVIEW

**Review by VikasKumarC — APPROVED** [review]

File: corio.py [hunk 1]
```diff
@@ -191,6 +193,7 @@ def schedule_execution_plan(parsed_input: dict, options: munch.Munch):
                       "endpoint_url": S3_CFG.endpoint,
                       "use_ssl": S3_CFG.use_ssl,
                       "seed": options.seed}
+    os.environ["sequential_run"] = str(options.sequential_run)
```
Comment (priyankaborawake) [inline corio.py]: Why this needs to be part of environment variable ? Any specific requirement.

Comment (DhobaleVishal) [reply corio.py]: We are using it in different modules, that's reason added it in environment variable.

Comment (priyankaborawake) [reply corio.py]: I don't think its a right way to do. 
We can pass it as function arguments.  For ex testplan.

File: src/commons/scheduler.py [hunk 2]
```diff
@@ -84,13 +92,15 @@ async def schedule_sessions(test_plan: str, test_plan_value: dict, common_params
...
-        task.result()
+    done, pending = await asyncio.wait(tasks, return_when=asyncio.FIRST_EXCEPTION)
+    if pending:
+        LOGGER.critical("Terminating process: %s & pending tasks: %s", process_name, pending)
+        for task in pending:
+            task.cancel()
+    if done:
+        LOGGER.info("Completed tasks: %s", done)
+        for task in done:
+            task.result()
```
Comment (ketanarlulkar) [inline src/commons/scheduler.py:102]: Please remove this for loop. It doesn't have any effect.

File: src/commons/report.py [hunk 3]
```diff
@@ -106,6 +107,8 @@ def update_tests_status(input_dict: dict, corio_start_time: datetime, value: dic
         if datetime.now() > (test_start_time + value['min_runtime']):
             pass_time = (test_start_time + value['min_runtime']).strftime('%Y-%m-%d %H:%M:%S')
             input_dict["RESULT_UPDATE"] = f"Passed at {pass_time}"
+            if ast.literal_eval(os.getenv("sequential_run")):
+                input_dict["TOTAL_TEST_EXECUTION"] = value['min_runtime']
```
Comment (priyankaborawake) [inline src/commons/report.py]: Total_test_execution wont be updated in case if its not sequential run ?

File: src/commons/scheduler.py (hunk 2 above)
Comment (DhobaleVishal) [reply src/commons/scheduler.py:102]: This is old code so I'll keep it as is.

File: src/commons/report.py (hunk 3 above)
Comment (DhobaleVishal) [reply src/commons/report.py]: It should work in either case. Added minor change in condition.

File: src/commons/support_bundle.py [hunk 4]
```diff
@@ -57,7 +57,8 @@ def collect_upload_rotate_support_bundles(dir_path: str, max_sb: int = 0) -> tup
             return False, f"Failed to collect support bundles for {nodes}."
         cluster_obj = ClusterServices(host, user, password)
         status, response = cluster_obj.collect_support_bundles(sb_dir)
-        assert status, IOError("Failed to generate support bundles. Response: %s", response)
+        if not status:
+            raise AssertionError(f"Failed to generate support bundles. Response: {response}")
```
Comment (ketanarlulkar) [inline src/commons/support_bundle.py:61]: All asserts are replaced with AssertionError. Any specific reason? asserts and raise AssertionError both do the same thing. Just that one extra line of code in raise.

Comment (DhobaleVishal) [reply src/commons/support_bundle.py:61]: Basically assert is used in tests not in libs.

File: src/commons/scheduler.py [hunk 5]
```diff
@@ -84,13 +93,15 @@ async def schedule_sessions(test_plan: str, test_plan_value: dict, common_params
...
         else:
             raise NotImplementedError(f"Tool is not supported: {each['tool']}")
-    done, pending = await asyncio.wait(tasks, return_when=asyncio.FIRST_COMPLETED)
-    LOGGER.critical("Terminating process & pending task: %s", process_name)
-    for task in pending:
-        task.cancel()
-    LOGGER.critical(done)
-    for task in done:
-        task.result()
+    done, pending = await asyncio.wait(tasks, return_when=asyncio.FIRST_EXCEPTION)
```
Comment (priyankaborawake) [inline src/commons/scheduler.py:94]: As discussed, process termination for sequential run is not handled in this PR.

**Review by ketanarlulkar — APPROVED** [review]
Approving as discussed.

File: src/commons/scheduler.py (hunk 5 above)
Comment (DhobaleVishal) [reply src/commons/scheduler.py:94]: This will be taken care in RVF task.

**Review by priyankaborawake — APPROVED** [review]
Approved.

**Review by akshaym99 — APPROVED** [review]

**Pull request merged**